### PR TITLE
chart, build, salt: bump loki chart to 6.20.0

### PR DESCRIPTION
**Pull request by eg-ayoub**

### CODE REVIEW

File: charts/loki.yaml [hunk 1]
```diff
@@ -8,19 +8,32 @@ loki:
     registry: ""
     repository: '__image__(loki)'
 
-  existingSecretForConfig: loki
+  # needs to be removed asap when https://github.com/grafana/loki/issues/14497 is fixed
+  schemaConfig:
+    remove: me
```
Comment (eg-ayoub) [inline charts/loki.yaml:13]: not sure what to do here , see issue

File: salt/metalk8s/addons/logging/loki/config/loki.yaml [hunk 2]
```diff
@@ -28,15 +26,21 @@ spec:
       chunk_retain_period: 1m
       wal:
         dir: /var/loki/loki/wal
+        # ingester automatically flushes to long term storage on shutdown
+        # so new ingester replays WAL instead of transfers.
```
Comment (eg-ayoub) [inline salt/metalk8s/addons/logging/loki/config/loki.yaml]: remove

File: salt/metalk8s/addons/logging/loki/config/loki.yaml [hunk 3]
```diff
@@ -28,15 +26,21 @@ spec:
...
+        flush_on_shutdown: true
       lifecycler:
         ring:
           kvstore:
             store: memberlist
-      max_transfer_retries: 0
     limits_config:
-      enforce_metric_name: false
+      # introduced in schema v13 (loki 3)
```
Comment (eg-ayoub) [inline salt/metalk8s/addons/logging/loki/config/loki.yaml]: remove

File: salt/metalk8s/addons/logging/loki/config/loki.yaml [hunk 4]
```diff
@@ -28,15 +26,21 @@ spec:
...
+      allow_structured_metadata: false
+      # enforce_metric_name: true DEPRECATED
```
Comment (eg-ayoub) [inline salt/metalk8s/addons/logging/loki/config/loki.yaml]: remove

File: salt/metalk8s/addons/logging/loki/config/loki.yaml [hunk 5]
```diff
@@ -28,15 +26,21 @@ spec:
...
       reject_old_samples: true
       reject_old_samples_max_age: 168h
+      # moved from chunk_store_config
```
Comment (eg-ayoub) [inline salt/metalk8s/addons/logging/loki/config/loki.yaml]: remove

**Review by m4nch0t — APPROVED** [review]

**Pull request merged**